guides/07_streaming/04_conversational-chatbot.md

```diff
@@ -0,0 +1,187 @@
+# Building Conversational Chatbots with Gradio
+
+Tags: AUDIO, STREAMING, CHATBOTS
+
+## Introduction
+
+The next generation of AI user interfaces is moving towards audio-native experiences. Users will be able to speak to chatbots and receive spoken responses in return. Several models have been built under this paradigm, including GPT-4o and [mini omni](https://github.com/gpt-omni/mini-omni).
+
+In this guide, we'll walk you through building your own conversational chat application using mini omni as an example. You can see a demo of the finished app below:
+
+[omni mini demo](https://github.com/user-attachments/assets/db36f4db-7535-49f1-a2dd-bd36c487ebdf)
+
+## Application Overview
+
+Our application will enable the following user experience:
+
+1. Users click a button to start recording their message
+2. The app detects when the user has finished speaking and stops recording
+3. The user's audio is passed to the omni model, which streams back a response
+4. After omni mini finishes speaking, the user's microphone is reactivated
+5. All previous spoken audio, from both the user and omni, is displayed in a chatbot component
+
+Let's dive into the implementation details.
+
+## Processing User Audio
+
+We'll stream the user's audio from their microphone to the server and determine if the user has stopped speaking on each new chunk of audio.
+
+Here's our `process_audio` function:
+
+```python
+import numpy as np
+from utils import determine_pause
+
+def process_audio(audio: tuple, state: AppState):
```

Comment:
Nice to use a dataclass for State, we should encourage this pattern elsewhere in the docs